File: 11/byungkyu/1334.cpp
class Solution {
 public:
  int dis[105][105];
  int findTheCity(int n, vector<vector<int>>& edges, int distanceThreshold) {
    for (int i = 0; i < n; i++) {
      for (int j = 0; j < n; j++) {
        dis[i][j] = 1e9;
      }
      dis[i][i] = 0;
    }
    for (auto& vec : edges) {
      int a = vec[0], b = vec[1], w = vec[2];
      dis[a][b] = dis[b][a] = w;
    }
    for (int k = 0; k < n; k++) {
      for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
          int nd = min(dis[i][j], dis[i][k] + dis[k][j]);
          dis[i][j] = dis[j][i] = nd;
        }
      }
    }
    int mi = 1e9, idx = -1;
    for (int i = 0; i < n; i++) {
      int c = 0;
      for (int j = 0; j < n; j++) {
        if (dis[i][j] <= distanceThreshold) c++;
      }
      if (mi >= c) {
        mi = c;
        idx = i;
      }
      printf("%d %d\n", i, c);
    }
    return idx;
  }
};

File: 11/junyeong/1334.rs
struct Solution {}

impl Solution {
    pub fn find_the_city(n: i32, edges: Vec<Vec<i32>>, distance_threshold: i32) -> i32 {
        let mut distance_memo = vec![vec![1e4 as i32 + 1; n as usize];n as usize];
        for i in 0..n as usize {
            distance_memo[i][i] = 0;
        }

        for edge in edges {
            let from = edge[0] as usize;
            let to = edge[1] as usize;
            let weight = edge[2];
            distance_memo[from][to] = weight;
            distance_memo[to][from] = weight;
        }

        for bridge in 0..n as usize {
            for from in 0..n as usize {
                for to in 0..n as usize {
                    distance_memo[from][to] = distance_memo[from][to].min(
                        distance_memo[from][bridge] + distance_memo[bridge][to]
                    );
                }
            }
        }

        let mut ans = 0;
        let mut least_move_count = n;
        for from in 0..n {
            let mut move_count = 0;
            for to in 0..n {
                if from != to && distance_memo[from as usize][to as usize] <= distance_threshold {
                    move_count += 1;
                }
            }
            if move_count <= least_move_count {
                least_move_count = move_count;
                ans = from;
            }
        }
        ans
    }
}
